packages/connect-query-core/src/connect-query-key.ts

```diff
@@ -18,13 +18,60 @@ import type {
   DescMethodUnary,
   DescService,
   MessageInitShape,
+  MessageShape,
 } from "@bufbuild/protobuf";
-import type { Transport } from "@connectrpc/connect";
-import type { SkipToken } from "@tanstack/query-core";
+import type { ConnectError, Transport } from "@connectrpc/connect";
+import type { DataTag, InfiniteData, SkipToken } from "@tanstack/query-core";
```

Comment:
`@connectrpc/connect-query` has a peer dependency on `@tanstack/react-query` 5.x. It looks like `DataTag` shipped with [v5.0.0](https://github.com/TanStack/query/releases/tag/v5.0.0). It was merged a week earlier in https://github.com/TanStack/query/pull/6119.

But `TError` on `DataTag` was added later, in https://github.com/TanStack/query/pull/8361. I think it's fair to update the peer dependency constraint. WDYT?

Reply:
Yep, makes sense to me. I'll update it.